pkg/node/statestore.go

```diff
@@ -24,3 +26,22 @@ func InitStateStore(log logging.Logger, dataDir string) (ret storage.StateStorer
 	}
 	return leveldb.NewStateStore(filepath.Join(dataDir, "statestore"), log)
 }
+
+const overlayKey = "overlay"
+
+// CheckOverlayWithStore checks the overlay is the same as stored in the statestore
+func CheckOverlayWithStore(overlay swarm.Address, storer storage.StateStorer) error {
+	var storedOverlay swarm.Address
+	err := storer.Get(overlayKey, &storedOverlay)
+	if err != nil {
+		if !errors.Is(err, storage.ErrNotFound) {
+			return err
+		}
+		return storer.Put(overlayKey, overlay)
+	}
+
+	if !storedOverlay.Equal(overlay) {
+		return errors.New("overlay changed")
```

Comment:
Just a suggestion. It would be good to include expected and found overlay addresses in the error message.

Reply:
fixed